explorer/frontend/utils/common.js

```diff
@@ -0,0 +1,329 @@
+import { TRANSACTION_CHART_TYPE } from '@/constants';
+
+/**
+ * Converts date to a readable text string.
+ * @param {string} dateStr - Date string.
+ * @param {Function} translate - Translation function.
+ * @param {object} config - Configuration object.
+ * @param {string} config.type - Date type 'local' or 'UTC'.
+ * @param {boolean} config.hasTime - Whether to include time.
+ * @param {boolean} config.hasSeconds - Whether to include seconds.
+ * @param {boolean} config.hasDays - Whether to include days.
+ * @returns {string} Formatted date.
+ * @example
+ * formatDate('2021-01-01T00:00:00.000Z', translate, { type: 'local', hasTime: true, hasSeconds: true, hasDays: true });
+ */
+export const formatDate = (dateStr, translate, config = {}) => {
+	const { type, hasTime = false, hasSeconds = false, hasDays = true } = config;
+	const months = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'];
+
+	const addZero = num => {
+		return 0 <= num && 10 > num ? '0' + num : num + '';
+	};
+
+	const dateObj = type === 'local' ? dateToLocalDate(dateStr) : new Date(dateStr);
+	const seconds = addZero(dateObj.getSeconds());
+	const minutes = addZero(dateObj.getMinutes());
+	const hour = addZero(dateObj.getHours());
+	const month = 'function' === typeof translate ? translate('month_' + months[dateObj.getMonth()]) : months[dateObj.getMonth()];
+	const day = dateObj.getDate();
+	const year = dateObj.getFullYear();
+
+	let formattedDate = `${month}`;
+	formattedDate += hasDays ? ` ${day}` : '';
+	formattedDate += `, ${year}`;
+	formattedDate += hasTime ? ` • ${hour}:${minutes}` : '';
+	formattedDate += hasTime && hasSeconds ? `:${seconds}` : '';
+
+	return formattedDate;
+};
+
+/**
+ * Converts date to a local date.
+ * @param {string} date - Date string.
+ * @returns {Date} Local date.
+ */
+export const dateToLocalDate = date => {
+	return new Date(new Date(date).getTime() - new Date(date).getTimezoneOffset() * 60000);
+};
```

Comment:
What does the 60000?

Reply:
`.getTime()` returns current timestamp in milliseconds and `.getTimezoneOffset()` returns the timezone offset in minutes, so we use `* 60000` to convert minutes to milliseconds.

Reply:
Thanks for the explanation.  Maybe creating a const for this?